### DISCUSSION TITLE
Authentication Error "The maximum number of teams has been reached"

### DISCUSSION THREAD

**Discussion by jamp-scp28** [outline/outline#3643] [Self Hosting] [answered]
I hope you are having a great day! first of all, thanks for the excellent work with this application.

We are trying to install the latest version of outline self-hosted on AWS, so far we managed to followed each of the steps, configuring at least one authentication provider, but we are having an issue with the callback.

The error occurs with any authentication provider (we tried slack and azure with our company domain), we are receiving the next error:

"The maximum numbers of teams has been reached", "stack": "UnauthorizedError" at "outline/build/server/commands/accountProvisioner.js:44:43".

We already try changing the api key of the authentication providers with another domain, deleting the database and reinstalling but so far not luck.

Thanks in advance if you can give us some insights on how to solve this problem.

**Comment by tommoor** [comment, marked as answer]
The current version 0.64.0 is limited to a single authentication method and a single team. The next version 0.65.0 will be able to have multiple authentication methods for a single team.

If you're seeing this error in the meantime it means that you already have one authentication method registered in the database and are trying to signin with a different one. Hope this clarifies!

**Reply by kaanklky** [reply]
Hello @tommoor,

I just installed outline@0.64.3 and I have same issue as @jamp-scp28 described but I only have 1 authentication method so I couldn't figure out why I can't login.

setup: standalone on GCP
auth provider: Google
```
{"error":"The maximum number of teams has been reached","stack":"UnauthorizedError: The maximum number of teams has been reached\n    at AuthenticationError (/root/outline/build/server/errors.js:35:34)\n    at accountProvisioner (/root/outline/build/server/commands/accountProvisioner.js:44:43)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Strategy._verify (/root/outline/build/server/routes/auth/providers/google.js:63:18)","level":"error","message":"Error during authentication"}
```

**Comment by okontsograda** [comment]
The suggested answer doesn't really give a way to help resolve the issue. I'm experiencing the same issue with using slack integration, and I haven't setup any other kind of authentication provider. I'm use the most recent stable build. @tommoor is there something we can do to flush the authentication table? I've tried to drop the database altogether, to no avail. 

Thanks in advance for any light you can shed on this issue.

Here's the exact error I'm getting:
`{"error":"The maximum number of teams has been reached","stack":"UnauthorizedError: The maximum number of teams has been reached\n    at AuthenticationError (/home/ubuntu/outline/build/server/errors.js:33:34)\n    at accountProvisioner (/home/ubuntu/outline/build/server/commands/accountProvisioner.js:44:43)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Strategy._verify (/home/ubuntu/outline/build/server/routes/auth/providers/slack.js:56:22)","level":"error","message":"Error during authentication"}`

When querying the tables for auth_providers, it appears to be empty:
`select * from authentication_providers;

 id | name | providerId | enabled | teamId | createdAt 
----+------+------------+---------+--------+-----------
(0 rows)`

**Reply by jamp-scp28** [reply]
Hi guys! I remember I managed to solve this issue by changing the logic in the source code, I do not remember exactly what I did; but maybe you can try finding the warning message in the source code and trying to change the maximum number to some different than 1.

After solving several errors including the maximum numbers of teams, I face a problem regarding websockets when the application was running on our servers using https, we could not find a solution for that one, ended up using wiki.js which turned out to be really good and suit my team needs.

Let me see if I can find some time to try to explain more on how to solve this problem, maybe today in the evening.

**Reply by okontsograda** [reply]
So it looks like I may have a different issue. I setup a new instance of outline and configured slack to work for the authentication. When I try to login, I get to the slack "Allow" redirect, and then when I select "Allow" I get re-directed to the login page and that error is what I see in my pm2 logs. 

From what I can see, the authentication_providers table doesn't get updated, and I don't have any additional logging to go off of to help resolve this.

**Comment by jamp-scp28** [comment]
Hi! just following up my previous comments on this. Here is a temporal solution while the problem gets fixed by the maintainers.

Find the next file in your outline installation, is located at outline/tree/main/server/commands/teamCreator.ts, here is the link in the project for your reference:

[outline](https://github.com/outline/outline)/[server](https://github.com/outline/outline/tree/main/server)/[commands](https://github.com/outline/outline/tree/main/server/commands)/teamCreator.ts

then replace these lines:

```javascript
if (env.DEPLOYMENT !== "hosted") {
    const team = await Team.findOne();

    // If the self-hosted installation has a single team and the domain for the
    // new team is allowed then assign the authentication provider to the
    // existing team
    if (team && domain) {
      if (await team.isDomainAllowed(domain)) {
        authP = await team.$create<AuthenticationProvider>(
          "authenticationProvider",
          authenticationProvider
        );
        return {
          authenticationProvider: authP,
          team,
          isNewTeam: false,
        };
      } else {
        throw DomainNotAllowedError();
      }
    }

    throw MaximumTeamsError();
  }
```

with the logic from an older version, and in the last lines change the maximum number of teams for a number > 1 and that should work.

```javascript
 if (_env.default.DEPLOYMENT !== "hosted") {
    const teamCount = await _models.Team.count(); // If the self-hosted installation has a single team and the domain for the
    // new team is allowed then assign the authentication provider to the
    // existing team

    if (teamCount === 1 && domain) {
      const team = await _models.Team.findOne();
      (0, _invariant.default)(team, "Team should exist");

      if (await team.isDomainAllowed(domain)) {
        authP = await team.$create("authenticationProvider", authenticationProvider);
        return {
          authenticationProvider: authP,
          team,
          isNewTeam: false
        };
      } else {
        throw (0, _errors.DomainNotAllowedError)();
      }
    }

     /*Replace this line with a number > 1 example 5 */ 
    
   if (teamCount >= 5) {
      throw (0, _errors.MaximumTeamsError)();
    }
  } 
```

I hope it helps you guys!
